homeassistant/components/ialarm_xr/__init__.py

```diff
@@ -18,18 +18,29 @@
     CONF_PASSWORD,
     CONF_PORT,
     CONF_USERNAME,
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_TRIGGERED,
     Platform,
 )
 from homeassistant.core import HomeAssistant
 from homeassistant.exceptions import ConfigEntryNotReady
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
-from .const import DOMAIN, IALARMXR_TO_HASS
+from .const import DOMAIN
 from .utils import async_get_ialarmxr_mac
 
 PLATFORMS = [Platform.ALARM_CONTROL_PANEL]
 _LOGGER = logging.getLogger(__name__)
 
+IALARMXR_TO_HASS = {
```

Comment:
This should be defined within the alarm control panel platform.

Reply:
IALARMXR_TO_HASS maps the int representing the status mode as detected by the library (that interacts with the alarm system) into the Home Assistant alarm status. You are probably referring to 

`ALARM_STATE_TO_HASS = {
    "disarmed": STATE_ALARM_DISARMED,
    "armed_home": STATE_ALARM_ARMED_HOME,
    "armed_away": STATE_ALARM_ARMED_AWAY,
    "triggered": STATE_ALARM_TRIGGERED,
}`

in alarm_contro_panel.py code, isn't it?

Reply:
The coordinator should return the integer enum representation, not the Home Assistant alarm control panel string state.

Please remove the conversion here.

https://github.com/home-assistant/core/blob/4a5679db08e8624ce383c2527ecb4e72799026f4/homeassistant/components/ialarm_xr/__init__.py#L93

That conversion should happen in the alarm control panel platform.

Reply:
Oh great @MartinHjelmare I didn't understand anything before! 😅 ...and I hope now yes!